infer_segmentation.py

```diff
@@ -51,30 +53,41 @@ def prepare_image(image, target_size_1d):
         for j in range(padded_image.shape[1] // target_size_1d):
             tiles[i].append(padded_image[i*target_size_1d:(i+1)*target_size_1d, j*target_size_1d:(j+1)*target_size_1d].copy())
 
-    # scale the images to be between 0 and 1
+    # scale the images to be between 0 and 1 if GV
     for i in range(len(tiles)):
         for j in range(len(tiles[i])):
             tiles[i][j] = tiles[i][j] * 1./255
     return tiles
 
 
-def overlay_predictions(prepared_tiles, preds, prediction_threshold):
+def overlay_predictions(prepared_tiles, preds, prediction_threshold, background_class, output_type):
     prediction_tiles = []
     for i in range(len(prepared_tiles)):
         prediction_tiles.append([])
         for j in range(len(prepared_tiles[i])):
             prediction_tiles[i].append(np.dstack((prepared_tiles[i][j], prepared_tiles[i][j], prepared_tiles[i][j])))
             prediction_tiles[i][j] = (prediction_tiles[i][j] * 255).astype(int)
 
-            above_threshold_mask = preds[i][j].max(axis=2) >= prediction_threshold
+            above_threshold_mask = preds[i][j].max(axis=2) > prediction_threshold
             best_class_by_pixel = preds[i][j].argmax(axis=2)
+            color_counter = 0
             for class_i in range(preds[i][j].shape[-1]):
                 above_threshold_and_best_class = above_threshold_mask & (best_class_by_pixel == class_i)
-                prediction_tiles[i][j][above_threshold_and_best_class] = class_colors[class_i % len(class_colors)]
+                if (background_class is not None) and (class_i == background_class):
+                    continue
+                else:
+                    if 'labels' in output_type:
+                        prediction_tiles[i][j][above_threshold_and_best_class] = int((color_counter + 1) *
+                                                                                     np.floor(255/preds[i][j].shape[-1]))
+                    else:
+                        prediction_tiles[i][j][above_threshold_and_best_class] = class_colors[color_counter
+                                                                                              % len(class_colors)]
+                    color_counter += 1
```

Comment:
is somewhat minor but I'd suggest changing this line `color_counter += 1` to `color_counter = (color_counter + 1) % len(class_colors)` and remove the mod from inside the indexing

(I'd have just done the edits but want to make sure I'm not missing a bigger effect here)

Reply:
okay, i wasn't exactly sure what the mode did (there before my edits), but i'll make the move

Reply:
@Josh-Joseph, this is the last thing to sure up. I made your changes but i'm not really sure of the effect of mod. Otherwise ready for final check

Reply:
got it - looks good. the mod thing is probably easiest for me to just explain in like 3 mins in person next week?

Reply:
okay sounds good